### PR TITLE
Created GameEngine and util ScreenListener

### PR DESCRIPTION
**Pull request by lsizemore8**
Created the code for the GameEngine based off of the module responsibilities listed in the Architecture Specification. Currently the code has many errors due to other methods currently being missing from the code, but those will be fixed as other pieces come in. 

Testing: Unable to test yet due to GDXRoot, GameCanvas, and LoadingMode not being implemented.

### CODE REVIEW

**Review by BryanGraeser — CHANGES_REQUESTED** [review]
update level controller before getting isComplete and isFailed.
some printing for isComplete and isFailed
change isComplete boolean status to isSuccess (to clarify difference from AssetState.COMPLETE)

**Comment by BryanGraeser** [comment]
What is lightPressed supposed to represent in the input controller? It seems to be a binary button press linked to shift_left, but shouldn't it be a scroll wheel determining the size of the light circle? IE scroll up increases the radius and scroll down decreases it?

File: core/src/com/fallenflame/game/GameEngine.java [hunk 1]
```diff
@@ -1,5 +1,384 @@
...
+     */
+    public void update(float delta) {
+        InputController input = InputController.getInstance();
+
+        if (input.didFlare()) {
+            level.createFlare(input.getMousePosition());
+        }
+        if(input.didLight()){
+            level.lightFromPlayer(levelJson);
+        }
```
Comment (BryanGraeser) [inline core/src/com/fallenflame/game/GameEngine.java]: level needs the lightradius from input passed in here

File: core/src/com/fallenflame/game/GameEngine.java [hunk 2]
```diff
@@ -1,5 +1,384 @@
...
+     */
+    public void update(float delta) {
+        InputController input = InputController.getInstance();
+
+        if (input.didFlare()) {
+            level.createFlare(input.getMousePosition());
+        }
+        if(input.didLight()){
+            level.lightFromPlayer(input.getLightRadius(), levelJson);
+        }
```
Comment (BryanGraeser) [inline core/src/com/fallenflame/game/GameEngine.java]: Why is levelJson also passed in here? This is unnecessary I think.

Comment (lsizemore8) [reply core/src/com/fallenflame/game/GameEngine.java]: Wasn't sure if you would need it. Can def remove

**Review by BryanGraeser — APPROVED** [review]
Just resolve merge conflict and gtg

**Pull request merged**